modules/account-lib/src/mpc/hdTree.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * An interface for calculating a subkey in an HD key scheme.
+ */
+const assert = require('assert');
+import { createHmac } from 'crypto';
+import { Ed25519Curve } from './curves';
+import { bigIntFromBufferBE, bigIntToBufferBE, bigIntFromBufferLE, bigIntToBufferLE } from './util';
+
+// 2^256
+const base = BigInt('0x010000000000000000000000000000000000000000000000000000000000000000');
+
+export interface Keychain {
+  key: any;
+  chaincode: any;
+}
+
+export default interface HDTree {
+
+  ready: Promise<any>;
+
+  publicDerive(keychain: Keychain, path: any): Keychain;
+
+  privateDerive(keychain: Keychain, path: any): Keychain;
```

Comment:
As this is an interface type, I thought it would be best for the path type to be implementation specific. For instance, you might have an HD scheme that only allows a single index derivation, in which case you can dynamically check for a number type path parameter.

Reply:
got it